src/components/EntityPage/Participant/Utils/Holder.js

```diff
@@ -0,0 +1,103 @@
+import React from "react";
+import {Container} from "../../../FileRepoSidebar/ui";
+import { Link } from 'react-router-dom';
+import styled from "react-emotion";
+import {withTheme} from "emotion-theming";
+
+//taken from secondaryNavMenu
+const MenuWrapper = styled('ul')`
+  ${({ theme }) => theme.secondaryNav}
+`;
+
+class Holder extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    //wrap the item in a one-element array if the item is not itself an array
+    if(!(props.children instanceof Array)) {
+      this.tabIDs = [props.children.props.label];
+      this.children = [props.children];
+    } else {
+      this.tabIDs = props.children.map( child => child.props.label );
+      this.children = props.children;
+    }
+
+    this.state = {activeTab: this.tabIDs[0]};
+  }
+
+  /**
+   * Makes the buttons for the tabs.
+   *
+   * @param holder This, as it is needed inside a nested class
+   * @returns {*|Array|*[]}
+   */
+  makeTabButtons(holder) {
+    return this.tabIDs.map(tabId => {
+
+      const isActive = tabId === this.state.activeTab;
+
+      //needs to be a class to use setState to do the hovering. Would do in CSS, but with emotion it's too complicated and hacky
+      class Button extends React.Component {
```

Comment:
Move the class outside of the `makeTabButtons` function, please, and pass `isActive` and `tabId` as props.

Classes should always be declared at the root of the module to avoid nasty nasty hoisting bugs.

Reply:
I double checked and I am wrong. Classes are not hoisted.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/class#Description

Yet, I'd rather have classes being declared at the top level of the module. It can't be used from outside the module if you don't `export` it.

Reply:
Okay! It's done.